service/src/internal/inclusion.rs

```diff
@@ -380,32 +372,30 @@ impl InclusionService {
         let (e, job_status);
         match zk_client_error {
             SP1NetworkError::SimulationFailed | SP1NetworkError::RequestUnexecutable { .. } => {
-                e = InclusionServiceError::DaClientError(format!(
+                e = InclusionServiceError::ZkClientError(format!(
                     "ZKP program critical failure: {zk_client_error} occurred for {job:?} PLEASE REPORT!"
                 ));
                 job_status = JobStatus::Failed(e.clone(), None);
             }
             SP1NetworkError::RequestUnfulfillable { .. } => {
-                e = InclusionServiceError::DaClientError(format!(
-                    "ZKP network failure: {zk_client_error} occurred  for {job:?} PLEASE REPORT!"
+                e = InclusionServiceError::ZkClientError(format!(
+                    "ZKP network failure: {zk_client_error} occurred for {job:?} PLEASE REPORT!"
                 ));
                 job_status = JobStatus::Failed(e.clone(), None);
             }
-            SP1NetworkError::RequestTimedOut { request_id } => {
-                e = InclusionServiceError::DaClientError(format!(
-                    "ZKP network: {zk_client_error} occurred  for {job:?}"
+            SP1NetworkError::RequestTimedOut { .. } => {
+                e = InclusionServiceError::ZkClientError(format!(
+                    "ZKP network: {zk_client_error} occurred for {job:?} - callback to start the job over"
                 ));
 
-                let id = request_id
-                    .as_slice()
-                    .try_into()
-                    .expect("request ID is always correct length");
+                // TODO: We cannot clone KeccakInclusionToDataRootProofInput thus we cannot insert into a JobStatus::DataAvailable(proof_input)
+                // So we just redo the work from scratch for the DA side as a stupid workaround
                 job_status =
-                    JobStatus::Failed(e.clone(), Some(JobStatus::ZkProofPending(id).into()));
+                    JobStatus::Failed(e.clone(), Some(JobStatus::DataAvailabilityPending.into()));
```

Comment:
this also sucks. not sure best path to not redo work here. We only hold the job status data until we move to the next phase, so we likely need to store stuff longer if we need to redo work from anything but the last step :thinking: